### DISCUSSION TITLE
improvemnet of junction restriction manager

### DISCUSSION THREAD

**Discussion by kianzarrin** [CitiesSkylinesMods/TMPE#1578] [General]
I want to simplify and expand the API of junction restriction manager. it will depend on https://github.com/CitiesSkylinesMods/TMPE/pull/1556

first I expose the enum in API : https://github.com/CitiesSkylinesMods/TMPE/blob/ee6cfee81a5721e1c14677404e24b53fb00cac94/TLM/TLM/Manager/Impl/JunctionRestrictionsManager.cs#L1321

Then I add these to the API:
```cs
bool IsAllowed(JunctionRestrictionFlags rule);
bool SetRule(JunctionRestrictionFlags rule, bool ?allowed); // null -> reset to default
void GetRule(JunctionRestrictionFlags rule, out bool? allowed, out bool default, out bool configurable);
void OverrideRule(JunctionRestrictionFlags rule, bool ?allowed, bool ?configurable); // null => remove override.
```

**Comment by Elesbaan70** [comment]
I am already working on this, and stashed my work when I realized that it was going to break NC2 and NCR. ~It actually works a bit differently from what you're describing, being very much built around bit masking.~ But it won't be possible to finish this until the NC2/NCR patches are gone.

Let's not duplicate work. 😬 😉

**Comment by Elesbaan70** [comment]
Actually, I went back and looked at the stash, and remembered that I decided the bit masking stuff was unnecessarily complicated. So what I'm doing isn't _very_ different from what you suggested. 🙂
